server/src/main/scala/org/http4s/server/middleware/authentication/Nonce.scala

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2014 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.server.middleware.authentication
+
+import org.http4s.crypto.unsafe.SecureRandom
+
+import java.math.BigInteger
+import java.util.Date
+
+private[authentication] class Nonce(val created: Date, var nc: Int, val data: String)
+
+private[authentication] object Nonce {
+  val random = new SecureRandom()
```

Comment:
When this becomes an effect, it can no longer be a global. So where will it be initialized?

Reply:
The usage is deprecated.  I don't think we need to make it an effect.  Though we could maybe add a deprecation notice to this type, to assure it's only called through deprecated code.

Reply:
Oh, is this not going to use the stuff from https://github.com/http4s/http4s/pull/6165?

Reply:
I was planning on welding it together after this.  @blast-hardcheese has suffered enough.  But I could merge it into this just to see what it looks like.

Reply:
Cool, sounds good to me 🚀